contracts/xsgd_contract.scilla

```diff
@@ -47,9 +47,7 @@ field masterMinter : ByStr20 = init_owner
 field paused : Bool = False
 field blacklister : ByStr20 = init_owner
 field blacklisted : Map ByStr20 Uint128 = Emp ByStr20 Uint128
-field balances : Map ByStr20 Uint128 = Emp ByStr20 Uint128
 field allowed : Map ByStr20 (Map ByStr20 Uint128) = Emp ByStr20 (Map ByStr20 Uint128)
-field totalSupply : Uint128  = zero
 field minters : Map ByStr20 Uint128 = Emp ByStr20 Uint128
```

Comment:
My understanding of this change is that `balances` and `total_supply` is moved to the proxy contract to ensure that data migration is not an issue when updating the token contract.

That raises the question as to why it is not necessary to migrate the other parts of the state? Why is it not necessary to migrate `minters`, `blacklisted` and `allowed`, for instance?

The logical end point for that line of reasoning is of course that you end up with just one contract which then cannot be updated. So I guess the question that needs to be answered is this: Which parts of the logic/state of this application do you want to be able to change, and which parts should be set in stone?

As far as I can tell every possible answer to that question comes with its own set of maintenance problems, so it's a question that needs to be carefully considered. However, not considering it does not mean that the issue goes away - it just means you might inadvertently settle on an answer you aren't happy with.